src/frontend/hooks/server/mediaSync.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery,
+} from '@tanstack/react-query';
+import {useApi} from '../../contexts/ApiContext';
+import {MediaSyncSetting} from '../../sharedTypes';
+
+export const MEDIA_SYNC_SETTING_KEY = 'media_sync_setting';
+
+export function convertMediaSyncSetting(isArchive: boolean): MediaSyncSetting {
+  return isArchive ? 'everything' : 'previews';
+}
+
+export function isArchiveDevice(value: MediaSyncSetting): boolean {
+  return value === 'everything';
+}
+
+export function useGetMediaSyncSetting() {
+  const api = useApi();
+
+  return useSuspenseQuery({
+    queryKey: [MEDIA_SYNC_SETTING_KEY],
+    queryFn: async () => {
+      const isArchive = await api.getIsArchiveDevice();
+      return convertMediaSyncSetting(isArchive);
+    },
+  });
+}
+
+export function useSetMediaSyncSetting() {
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: async (newSetting: MediaSyncSetting) => {
+      const isArchive = isArchiveDevice(newSetting);
+      await api.setIsArchiveDevice(isArchive);
+    },
+    onMutate: async newSetting => {
+      await queryClient.cancelQueries({queryKey: [MEDIA_SYNC_SETTING_KEY]});
+      const previousSetting = queryClient.getQueryData<MediaSyncSetting>([
+        MEDIA_SYNC_SETTING_KEY,
+      ]);
+      queryClient.setQueryData([MEDIA_SYNC_SETTING_KEY], newSetting);
+      return {previousSetting};
+    },
+    onError: (err, newSetting, context) => {
+      if (context?.previousSetting) {
+        queryClient.setQueryData(
+          [MEDIA_SYNC_SETTING_KEY],
+          context.previousSetting,
+        );
+      }
+    },
```

Comment:
I think it would be simpler (and more maintainable) to [optimistic update via the ui](https://tanstack.com/query/latest/docs/framework/react/guides/optimistic-updates#via-the-ui)

You could remove all this code.

And in the component you could simply

```ts
const {mutate: setMediaSyncSetting, variable, isPending} = useSetMediaSyncSetting();

<SelectOne
        value={isPending ? variable: mediaSyncSetting}
        {...restProps}
      />

```